File: Regula Falsi/RegulaFalsi.c
//C Program to find the root of x*x*x-2*x-5=0 using Regula Falsi Method
#include<stdio.h>
#include<math.h>
#define EPSILON 0.00001
double f(double x){
    return x*x*x-2*x-5;
}
void regulaFalsi(double a,double b){
    if(f(a)*f(b)>=0){
        printf("You have not assumed right a and b\n");
        return;
    }
    double c=a;
    while(fabs(f(c))>=EPSILON){
        c=(a*f(b)-b*f(a))/(f(b)-f(a));
        if(f(c)==0.0)
            break;
        else if(f(c)*f(a)<0)
            b=c;
        else
            a=c;
    }
    printf("The value of root is : %lf\n",c);
}
int main(){
    double a=2,b=3;
    regulaFalsi(a,b);
    return 0;
}

File: Regula Falsi/RegulaFalsi.js
//Regula Falsi Method to find the root of x*x*x-2*x-5=0 Program in JavaScript
let E = 0.0001;
const fn = (x) => x * x * x - 2 * x - 5;
const regulaFalsi = (a, b) => {
    let c = a;
    while (Math.abs(fn(c)) >= E) {
        c = (a * fn(b) - b * fn(a)) / (fn(b) - fn(a));
        if (fn(c) == 0) break;
        else if (fn(c) * fn(a) < 0) b = c;
        else a = c;
    }
    console.log(`The value of root is : ${c}`);
}
let a = 2;
let b = 3;
regulaFalsi(a, b);

